rocAL/rocAL/source/augmentations/node_ssd_random_crop.cpp

```diff
@@ -182,28 +172,25 @@ void SSDRandomCropNode::update_node()
             auto left = crop_box.l, top = crop_box.t, right = crop_box.r, bottom = crop_box.b;
             for (int j = 0; j < bb_count; j++)
             {
-                int m = j * 4;
-                auto x_c = 0.5f * (2 * coords_buf[m] + coords_buf[m + 2]);
-                auto y_c = 0.5f * (2 * coords_buf[m + 1] + coords_buf[m + 3]);
+                auto x_c = 0.5f * (2 * coords_buf[j].l + coords_buf[j].r);
+                auto y_c = 0.5f * (2 * coords_buf[j].t + coords_buf[j].b);
                 if ((x_c >= left) && (x_c <= right) && (y_c >= top) && (y_c <= bottom))
                     valid_bbox_count++;
             }
             if (valid_bbox_count == 0)
                 continue;
             break;
         } // while loop
-        _x1_val[i] = (crop_box.l) * in_width[i];
-        _y1_val[i] = (crop_box.t) * in_height[i];
-        _crop_width_val[i] = (crop_box.r - crop_box.l) * in_width[i];
-        _crop_height_val[i] = (crop_box.b - crop_box.t) * in_height[i];
-        _x2_val[i] =  (crop_box.r) * in_width[i];
-        _y2_val[i] =  (crop_box.b) * in_height[i];
-
+        crop_dims[i].x1 = (crop_box.l) * input_roi[i].x2;
+        crop_dims[i].y1 = (crop_box.t) * input_roi[i].y2;
+        if(_inputs[0]->info().roi_type() == RocalROIType::XYWH) {
```

Comment:
probably you need to define rocalROI as a union which can have x1,y1,x2, y2 or x,y,w,h so the code for setting and getting the values is more readable

Reply:
Will issue a separate small PR for this change